File: backend/app/models/journal.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Boolean
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base


class Journal(Base):
    __tablename__ = 'journals'

    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, index=True, nullable=False)
    description = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, onupdate=datetime.utcnow)

    # ── Publication identity (JG-101) ────────────────────
    # One source of truth for masthead, DOI metadata, Scholar tags,
    # citation export and the public footer. Every field except licence
    # is nullable — a fresh journal fills these in as they are earned.
    issn_online = Column(String(20), nullable=True)
    issn_print = Column(String(20), nullable=True)
    abbreviation = Column(String(100), nullable=True)
    subject_area = Column(String(200), nullable=True)
    language = Column(String(50), nullable=True)
    start_year = Column(Integer, nullable=True)
    frequency = Column(String(100), nullable=True)
    publisher_name = Column(String(200), nullable=True)
    publisher_address = Column(Text, nullable=True)
    licence = Column(String(50), nullable=False, default='CC-BY-4.0')
    doi_prefix = Column(String(50), nullable=True)
    oai_identifier_prefix = Column(String(200), nullable=True)

    # Which record represents the currently-active JGAIR journal.
    # Only one row should carry is_active=True; GET /journals/current returns it.
    is_active = Column(Boolean, nullable=False, default=False)

    # ── Contact block (added by h4d8e5f6a2c1) ─────────────
    # Rendered by the public ContactPage sidebar when populated.
    phone = Column(String(50), nullable=True)
    address = Column(String(500), nullable=True)
    twitter_url = Column(String(300), nullable=True)
    linkedin_url = Column(String(300), nullable=True)
    email_editorial = Column(String(200), nullable=True)
    email_publisher = Column(String(200), nullable=True)

    articles = relationship("Article", back_populates="journal")

File: frontend/src/api/journal.ts
import client from './client';

// JG-101 — publication identity, the masthead source of truth.
export interface JournalIdentity {
    id: number;
    title: string;
    description: string | null;
    issn_online: string | null;
    issn_print: string | null;
    abbreviation: string | null;
    subject_area: string | null;
    language: string | null;
    start_year: number | null;
    frequency: string | null;
    publisher_name: string | null;
    publisher_address: string | null;
    licence: string;
    doi_prefix: string | null;
    oai_identifier_prefix: string | null;
    // ── Contact block (migration h4d8e5f6a2c1) ────────────
    // Populated via the editor's journal-identity form; consumed by the
    // public Contact page sidebar and the site footer.
    phone: string | null;
    address: string | null;
    twitter_url: string | null;
    linkedin_url: string | null;
    email_editorial: string | null;
    email_publisher: string | null;
    is_active: boolean;
}

export type JournalIdentityPatch = Partial<Omit<JournalIdentity, 'id' | 'is_active'>>;

export async function getCurrentJournal(): Promise<JournalIdentity> {
    const res = await client.get<JournalIdentity>('/journals/current');
    return res.data;
}

export async function updateCurrentJournal(
    patch: JournalIdentityPatch
): Promise<JournalIdentity> {
    const res = await client.patch<JournalIdentity>('/journals/current', patch);
    return res.data;
}
